### PR TITLE
:tada: show chart diffs with changed checksums

### PR DESCRIPTION
**Pull request by Marigold**
## Motivation

Chart-diff currently only shows charts with changed config (due to updating variable ids by indicator upgrader). We'd like to have an option to show all charts with changes - either coming from change to the underlying data or metadata.

## Summary

Use data and metadata checksums to optionally show chart differences, in addition to just changed chart configurations. This information is displayed as tags next to a chart slug. Currently, only charts with changed configurations are shown by default, and approving them is enough for owidbot to give you a ✅. However, if desired, you can enable the option to show data and metadata changes in the filters.

The new function `modified_charts_by_admin` assumes that all charts or datasets were created by an Admin user with ID=1. This is a temporary solution until we determine how to display the real name of the creator.

I think we should carefully test this before making it the default. I am concerned that updating datasets, such as regions or population, would trigger a review for almost all charts.

image

image

## How to test it

Go to chart-diff on the staging server and select "metadata" in filters. You should see a cherry blossom chart with updated metadata.

@lucasrodes I noticed that warnings have been turned off. Is that expected? I want to show a warning if `GRAPHER_USER_ID` isn't Admin.

## TODO before merging:
- [x] undo modifications to cherry blossom

### CODE REVIEW

File: apps/wizard/app_pages/chart_diff/chart_diff.py [hunk 1]
```diff
@@ -78,7 +87,7 @@ def slug(self) -> str:
         """
         if self.target_chart:
             assert self.source_chart.config["slug"] == self.target_chart.config["slug"], "Slug mismatch!"
-        return cast(str, self.source_chart.config["slug"])
+        return self.source_chart.config.get("slug", "no-slug")
```
Comment (lucasrodes) [inline apps/wizard/app_pages/chart_diff/chart_diff.py:90]: when is the case that a chart won't have a slug?

File: etl/steps/data/garden/biodiversity/2024-01-25/cherry_blossom.meta.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 dataset:
-  update_period_days: 365
+  update_period_days: 368
```
Comment (lucasrodes) [inline etl/steps/data/garden/biodiversity/2024-01-25/cherry_blossom.meta.yml:2]: is this value change expected? or was it only to test things out?

Comment (Marigold) [reply etl/steps/data/garden/biodiversity/2024-01-25/cherry_blossom.meta.yml:2]: Testing things, I'll remove it before merging.

File: apps/wizard/app_pages/chart_diff/app.py [hunk 3]
```diff
@@ -47,8 +48,15 @@
...
-# st.warning("- " + "\n\n- ".join(warn_msg))
+# WARN_MSG += ["This tool is being developed! Please report any issues you encounter in `#proj-new-data-workflow`"]
+
+if config.GRAPHER_USER_ID != 1:
+    WARN_MSG.append(
+        "Your `GRAPHER_USER_ID` is not set to 1 (Admin). Please start using STAGING or change it in your env file."
+    )
+
+if WARN_MSG:
+    st.warning("- " + "\n\n- ".join(WARN_MSG))
```
[the remark this answers is not in the document]
Comment (Marigold) [reply apps/wizard/app_pages/chart_diff/app.py]: Added more info.

File: apps/wizard/app_pages/chart_diff/chart_diff.py (hunk 1 above)
Comment (Marigold) [reply apps/wizard/app_pages/chart_diff/chart_diff.py:90]: I created a new draft chart for testing and it didn't have slug for some reason. It's an edge case which won't happen in real life, but I had to fix it for development.

**Pull request merged**